### PR TITLE
docs: Update installation instructions for pnpm

### PR DESCRIPTION
**Pull request by tylerbutler**
This is a companion PR to #13554. This PR contains only documentation updates to reflect that we use pnpm for package management in the client release group (after the other PR is merged). I split it out to make the PRs easier to review.

### CODE REVIEW

File: build-tools/packages/build-tools/src/bumpVersion/README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Bump Version tool
+# Bump Version tool (deprecated)
```
Comment (alexvy86) [inline build-tools/packages/build-tools/src/bumpVersion/README.md:1]: Not sure this was part of this PR? It's docs, I don't feel strongly about moving it, just pointing it out.

File: examples/hosts/app-integration/external-data/README.md [hunk 2]
```diff
@@ -16,15 +16,26 @@ TBD
...
 ## Getting Started
 
 You can run this example using the following steps:
 
-1. Run `npm install` and `npm run build:fast -- --nolint` from the `FluidFramework` root directory.
+1. Install [pnpm](https://pnpm.io/) by running `npm i -g pnpm`.
+1. Run `pnpm install` and `npm run build:fast -- --nolint` from the `FluidFramework` root directory.
     - For an even faster build, you can add the package name to the build command, like this:
       `npm run build:fast -- --nolint @fluid-example/app-integration-external-data`
+1. In a separate terminal, start a Tinylicious server by following the instructions in [Tinylicious](../../../server/tinylicious).
```
Comment (alexvy86) [inline examples/hosts/app-integration/external-data/README.md:33]: I think this line is unnecessary and the note in the previous version is correct: `"start": "start-server-and-test start:tinylicious 7070 start:app",`

**Review by alexvy86 — COMMENTED** [review]
The docs for the `azure/` folder are also updated. Is the plan to move that one to `pnpm` soon as well and then docs for that one together with the client release group?

File: examples/hosts/app-integration/external-data/README.md (hunk 2 above)
Comment (tylerbutler) [reply examples/hosts/app-integration/external-data/README.md:33]: Good catch; fixed.

**Comment by tylerbutler** [comment]
> The docs for the `azure/` folder are also updated. Is the plan to move that one to `pnpm` soon as well and then docs for that one together with the client release group?

This wasn't intentional, and is a side effect of our templatized readme system. I could add another option to the readme generation, but instead, I'll push to get the azure pnpm PR ready to go (#13094).

**Review by Josmithr — APPROVED** [review]
🚀

**Pull request merged**